vpc/service/db/migrations/30_branch_eni_sg_check.up.sql

```diff
@@ -0,0 +1,46 @@
+START TRANSACTION ;
+CREATE OR REPLACE FUNCTION check_sg_exists() RETURNS trigger
+    LANGUAGE plpgsql AS
+$$
+DECLARE
+    security_group_id text;
+BEGIN
+    FOREACH security_group_id IN ARRAY NEW.security_groups
+        LOOP
+            PERFORM id FROM security_groups WHERE group_id = security_group_id;
+            IF NOT FOUND THEN
+                RAISE EXCEPTION 'Cannot find security group %', security_group_id;
+            end if;
+        END LOOP;
+    RETURN NEW;
+END;
+$$;
+
+CREATE CONSTRAINT TRIGGER sg_exists_on_update AFTER UPDATE OF security_groups ON branch_enis DEFERRABLE
+    FOR EACH ROW EXECUTE PROCEDURE check_sg_exists();
+
+CREATE CONSTRAINT TRIGGER sg_exists_on_insert AFTER INSERT ON branch_enis DEFERRABLE
+    FOR EACH ROW EXECUTE PROCEDURE check_sg_exists();
+
+
+CREATE OR REPLACE FUNCTION check_sg_modifications() RETURNS trigger
+    LANGUAGE plpgsql AS
+$$
```

Comment:
missing `;`? jk I have no idea when you need semicolons here, or even what `$$` means :)